### PR TITLE
Cabal build files -> hpack/stack built build files & update GHC version to Stack GHC 8.8.4: Resolve JacquesCarette/Drasil#2292 stack packages

### PR DESCRIPTION
**Pull request by balacij**

### CODE REVIEW

File: code/drasil-lang/Language/Drasil/URI/AST.hs [hunk 1]
```diff
@@ -17,10 +17,10 @@ data Authority = Full Username Password Host Port
                                   
 type Path = String     -- ^ Type the full path excluding the first /
 type Query = [String]  -- ^ Make sure [] doesn't print a "?"
-                       ---Separate elements with "&"
-type Fragment = String ---Make sure "" becomes "" not "#" when printing
-type Username = String ---Again make sure "" doesn't print anything
-type Password = String ---Don't print anything if "" or if Username=""
-type Host = String     -- | Type the host address (ex. "www.github.com")
```
Comment (JacquesCarette) [inline code/drasil-lang/Language/Drasil/URI/AST.hs:24]: Hmmm, I'm pretty sure those comments are special for Haddock, and that's why they had a `|` in them.

**Review by JacquesCarette — APPROVED** [review]

**Comment by JacquesCarette** [comment]
So this is ready to roll in?  [The automatic builds didn't seem to kick in... we might have run out of tokens. It's going to be urgent to get these switch over to github actions!]

File: code/drasil-lang/Language/Drasil/URI/AST.hs (hunk 1 above)
Comment (balacij) [reply code/drasil-lang/Language/Drasil/URI/AST.hs:24]: Ah, yes. I forgot to mention this change, my apologies.

When I run `stack haddock`, I was getting this error:
```
drasil-lang    > Language/Drasil/URI/AST.hs:24:24: error:
drasil-lang    >     parse error on input ‘-- | Type the host address (ex. "www.github.com")’
drasil-lang    >    |
drasil-lang    > 24 | type Host = String     -- | Type the host address (ex. "www.github.com")
drasil-lang    >    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
drasil-lang    > Haddock coverage:
drasil-lang    > Warning: 'tables' is out of scope.
drasil-lang    >     If you qualify the identifier, haddock can try to link it anyway.
drasil-lang    > Warning: 'Label' is out of scope.
drasil-lang    >     If you qualify the identifier, haddock can try to link it anyway.
Progress 1/12

--  While building package drasil-lang-0.1.60.0 (scroll up to its section to see the error) using:
      /home/jasonbalaci/.stack/setup-exe-cache/x86_64-linux-tinfo6/Cabal-simple_mPHDZzAJ_3.0.1.0_ghc-8.8.4 --builddir=.stack-work/dist/x86_64-linux-tinfo6/Cabal-3.0.1.0 haddock --html --hoogle --html-location=../$pkg-$version/ --haddock-option=--hyperlinked-source --haddock-option=--quickjump
```

So I "temporarily" (but I forgot, sorry) erased it. This actually brings up a question I had as well -- where is the Haddock documentation for this file? It doesn't appear to be on https://jacquescarette.github.io/Drasil/docs/index.html .

Comment (JacquesCarette) [reply code/drasil-lang/Language/Drasil/URI/AST.hs:24]: I don't know why it's not there. It looks like quite a few files are missing. Perhaps they don't have any Haddock documentation, or something else fails for them. Can you create an issue to see what's going on?

Note that if Haddock fails, it's a good reason to not have a comment be mean for Haddock to deal with! [It should be done in a separate PR, normally, but it's ok not to worry about that right now.]

**Review by JacquesCarette — APPROVED** [review]

File: code/drasil-lang/Language/Drasil/URI/AST.hs (hunk 1 above)
Comment (balacij) [reply code/drasil-lang/Language/Drasil/URI/AST.hs:24]: Thank you, I created ticket #2294 for us to follow up.

**Comment by JacquesCarette** [comment]
So I've merged it... let's see if it all works!

**Comment by balacij** [comment]
Ah, thank you!

**Pull request merged**